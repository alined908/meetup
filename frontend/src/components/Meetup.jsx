import React, {Component} from 'react'
import Restauraunt from "./Restauraunt"
import {connect} from 'react-redux';
import {deleteMeetup} from '../actions/meetup';
import {getFriends} from "../actions/friend"
import Button from '@material-ui/core/Button';
import moment from 'moment';
import MeetupFriend from "./MeetupFriend"

class Meetup extends Component {
    constructor(props){
        super(props)
        if(!this.props.isFriendsInitialized){
            this.props.getFriends()
        }

    }
    
    handleDelete = (uri) => {
        this.props.deleteMeetup(uri);
    }

    render () {
        var choices = {}
        var valid = false;
        const [id, name, uri, location, datetime, options, chosen, members] = this.props.meetup
        if (options !== ""){
            choices = JSON.parse(options);
            valid = true
            console.log(choices)
        }

        const isMember = (friend) => {
            if (friend in members){
                return true
            } else {
                return false
            }
        }

        const renderFriends = () => {
            return (
                <div>
                    {this.props.isFriendsInitialized && this.props.friends.map((friendship) => <MeetupFriend friend={friendship.user} isMember={isMember(friendship.user.id)}></MeetupFriend>)}
                </div>
            )
        }
        
        return (
            <div className="meetup">
                <div>{name}</div>
                <div>{moment(datetime).local().format("MMM DD h:mm A")}</div>
                <div>{location}</div>
                <div>Members - {Object.keys(members).map((key) => members[key].email + " ")}</div>
                <Button variant="contained" color="secondary" onClick={() => this.handleDelete(uri)}>Delete Meetup</Button>
                <div className="rsts">
                    {valid && choices.businesses.map((rest) => 
                        <Restauraunt data={rest}/>
                    )}
                </div>
                <div>{renderFriends()}</div>
            </div>
        )
    }
}

function mapStateToProps(state){
    return {
       friends: state.user.friends,
       isFriendsInitialized: state.user.isFriendsInitialized
    }
}

const mapDispatchToProps = {
    deleteMeetup,
    getFriends
}

export default connect(mapStateToProps, mapDispatchToProps)(Meetup)