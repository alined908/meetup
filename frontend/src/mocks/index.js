import moment from 'moment'

//Category
export const category = {
    id: 1, 
    image: null,
    label: "Dessert", 
    api_label: "desserts"
}

//User
export const user = {
    id: 1, 
    email: "daniel@gmail.com", 
    first_name: "Daniel", 
    last_name: "Lee",
    avatar: null
}
export const user2 = {
    id: 2, 
    email: "example2@gmail.com", 
    first_name: "Bob", 
    last_name: "Jim", 
    avatar: null
}
export const user3 = {
    id: 3, 
    email: "example3@gmail.com", 
    first_name: "Bob", 
    last_name: "Jim", 
    avatar: null
}
export const settings = {
    radius: 25, 
    location: null, 
    latitude: null,
    longitude: null
}

export const userWithSettings = {
    ...user,
    settings: {...settings}
}
export const friend = {
    id: 1, 
    user: user2, 
    created_at: moment().format(), 
    chat_room: "something"
}
export const friends = [friend]
export const preference = {
    id: 1, 
    user, 
    category, 
    name: "Dessert", 
    ranking: 1, 
    timestamp: moment().format()
}
export const preferences = [preference]
export const friendInvite = {
    id: 1, 
    timestamp: moment().format(), 
    status: 1, 
    uri: "random", 
    sender: user, 
    receiver: user2
}

//Chat
export const message = {
    id: 1, 
    message: "hello", 
    timestamp: moment().format(), 
    room_id: 1, 
    sender: user, 
    is_notif: false
}
export const message2 = {
    id: 2, 
    message: "hello", 
    timestamp: moment().format(), 
    room_id: 1, 
    sender: user, 
    is_notif: true
}
export const messages = [message, message2]
export const members = {1: user, 2: user2}
export const room = {
    uri: {
        id: 1, 
        uri: "abc", 
        name: "Room", 
        timestamp: moment().format(), 
        members: members, 
        friendship: 1, 
        meetup:null, 
        notifs: 0
    }
}
export const rooms = {
    friend: {
        id: 1, 
        uri: "abc", 
        name: "Room", 
        timestamp: moment().format(), 
        members: members, 
        friendship: 1, 
        meetup:null, 
        notifs: 0
    },
    meetup: {
        id: 2, 
        uri: "xyz", 
        name: "Room", 
        timestamp: moment().format(), 
        members: members, 
        friendship: null, 
        meetup:1, 
        notifs: 3
    }
}

//General
export const error = {
    message: "Something went wrong."
}

//Meetup
export const meetupMember1 = {
    1: {
        id: 1,
        user,
        ban: false,
        admin: true
    }
}

export const meetupMember2 = {
    2: {
        id: 2,
        user: user2,
        ban: false,
        admin: false
    }
}

export const meetupMember3 = {
    3: {
        id: 3,
        user: user3,
        ban: false,
        admin: false
    }
}

export const meetupMembers = {...meetupMember1, ...meetupMember2}

export const meetup = {
    uri: {
        id: 1, 
        name: "Meetup", 
        uri: "uri", 
        creator: user, 
        location: "location", 
        date: moment().format("YYYY-MM-DD"),
        members: meetupMembers,
        notifs: 0,
        public: true,
        categories: [category],
        latitude: 34.228754, 
        longitude: -118.2351192,
        events: {},
        isMeetupEventsInitialized : true
    }
}

export const meetups = {...meetup}

export const event = {
    1: {
        id: 1,
        meetup: 1,
        creator: user,
        title: "Dessert",
        start: moment().format(),
        end: moment().format(),
        chosen: null,
        categories: [category],
        options: {},
        price: "1, 2",
        distance: 16000,
        entries: {},
        random: true
    }
}
export const events = {...event}

export const option = {
    1 : {
        id: 1,
        score: 0,
        option: null,
        restaurant: {},
        votes: {},
        banned: false
    }
}

export const options = {...option}

export const tokens = {
    access: "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ0b2tlbl90eXBlIjoiYWNjZXNzIiwiZXhwIjoxNTg4MjI5OTEzLCJqdGkiOiIwOTg2YTg0MDA1ZGM0OGQ1ODEzYmIyNzE1OWViNGUyZCIsInVzZXJfaWQiOjEsInVzZXIiOnsiaWQiOjEsImVtYWlsIjoiZGFuaWVsQGdtYWlsLmNvbSIsImZpcnN0X25hbWUiOiJEYW5pZWwiLCJsYXN0X25hbWUiOiJMZWUiLCJhdmF0YXIiOm51bGwsInNldHRpbmdzIjp7InJhZGl1cyI6MjUsImxvY2F0aW9uIjpudWxsLCJsYXRpdHVkZSI6bnVsbCwibG9uZ2l0dWRlIjpudWxsfX19.FbffRthOKX5F0TGvqA5mZXuXUWNkUojzKexDGCoP518",
    refresh : "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ0b2tlbl90eXBlIjoicmVmcmVzaCIsImV4cCI6MTU4ODM4Mjk0NiwianRpIjoiN2ZhNGY2MDE4OWM3NDA5ZmFjOTMwYmE4ZTRlMWFhMTMiLCJ1c2VyX2lkIjoxLCJ1c2VyIjp7ImlkIjoxLCJlbWFpbCI6ImRhbmllbEBnbWFpbC5jb20iLCJmaXJzdF9uYW1lIjoiRGFuaWVsIiwibGFzdF9uYW1lIjoiTGVlIiwiYXZhdGFyIjpudWxsLCJzZXR0aW5ncyI6eyJyYWRpdXMiOjI1LCJsb2NhdGlvbiI6bnVsbCwibGF0aXR1ZGUiOm51bGwsImxvbmdpdHVkZSI6bnVsbH19fQ.qqaY4Ye7rmyOKLbDS2EKz4SGWH0fwZ0MvrGP6NJ3IRE"
}