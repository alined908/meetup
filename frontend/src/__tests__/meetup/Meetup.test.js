import React from "react";
import { shallow, mount } from "enzyme";
import Meetup, { UnderlyingMeetup } from "../../components/meetup/Meetup";
import WebSocketService from "../../accounts/WebSocket";
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import * as mocks from "../../mocks";
import setupStore from "../../setupTests";

jest.mock("../../accounts/WebSocket");

const props = {
  meetup: mocks.meetup.uri,
  user: mocks.user,
  friends: mocks.friends,
  isFriendInitialized: false,
  isMeetupEventsFetching: true,
  isMeetupEventsInitialized: false,
  deleteMeetup: jest.fn(),
  getFriends: jest.fn(),
  getMeetupEvents: jest.fn(),
  addMeetupEvent: jest.fn(),
  reloadMeetupEvent: jest.fn(),
  voteMeetupEvent: jest.fn(),
  decideMeetupEvent: jest.fn(),
  deleteMeetupEvent: jest.fn(),
  sendMeetupEmails: jest.fn(),
  removeNotifs: jest.fn(),
  addMeetupMember: jest.fn(),
  addGlobalMessage: jest.fn(),
  addEventOption: jest.fn(),
  sendFriendInvite: jest.fn(),
  deleteEventOption: jest.fn(),
  deleteMeetupMember: jest.fn(),
  handlePublicMeetupJoin: jest.fn(),
  handleLeaveMeetup: jest.fn(),
};

describe("Meetup tests general", () => {
  const addMeetupCallbacks = jest.fn(),
    connect = jest.fn(),
    disconnect = jest.fn(),
    addChatCallbacks = jest.fn();
  window.confirm = jest.fn(() => true);

  WebSocketService.mockImplementation(() => {
    return {
      addMeetupCallbacks,
      connect,
      disconnect,
      addChatCallbacks,
    };
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("renders without crashing given props", () => {
    const wrapper = shallow(<UnderlyingMeetup {...props} />);
    expect(addMeetupCallbacks.mock.calls.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });

  it("should connect to socket and get relevant info", () => {
    const wrapper = shallow(<UnderlyingMeetup {...props} />);
    expect(connect.mock.calls.length).toBe(1);
    expect(props.getMeetupEvents.mock.calls.length).toBe(1);
    expect(props.getFriends.mock.calls.length).toBe(1);
    expect(props.removeNotifs.mock.calls.length).toBe(0);
  });

  it("should disconnect socket on unmount", () => {
    const wrapper = shallow(<UnderlyingMeetup {...props} />);
    wrapper.unmount();
    expect(disconnect.mock.calls.length).toBe(1);
  });

  it("should show location of meetup", () => {
    const wrapper = shallow(<UnderlyingMeetup {...props} />);
    const location = wrapper.find('[aria-label="location"]');
    expect(location).toHaveLength(1);
    expect(location.text()).toContain("location");
  });

  it("should show fetching state if meetupeventsfetching", () => {
    const wrapper = shallow(<UnderlyingMeetup {...props} />);
    expect(wrapper.find(".loading")).toHaveLength(1);
  });

  it("should show meetup events if meetupsevents not fetchng and initialized", () => {
    const newProps = {
      ...props,
      meetup: { ...props.meetup, events: { ...mocks.events } },
      isMeetupEventsFetching: false,
      isMeetupEventsInitialized: true,
    };
    const wrapper = shallow(<UnderlyingMeetup {...newProps} />);
    expect(wrapper.find(".loading")).toHaveLength(0);
    expect(wrapper.find("Connect(MeetupEvent)")).toHaveLength(1);
  });
});

describe("Meetup tests where user is creator", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should send deleteMeetup action if button clicked", () => {
    const wrapper = shallow(<UnderlyingMeetup {...props} />);
    const button = wrapper.find('[aria-label="delete"]');
    button.simulate("click");
    expect(props.deleteMeetup.mock.calls.length).toBe(1);
  });

  // it('should handle disabled email if button is clicked', () => {

  //     const store = setupStore()
  //     const wrapper = mount(
  //         <Provider store={store}>
  //             <Router>
  //                 <UnderlyingMeetup {...props}/>
  //             </Router>
  //         </Provider>
  //     )
  //     const emailButton = wrapper.find('ProgressIcon')
  //     console.log(wrapper.debug())
  //     expect(emailButton).toHaveLength(1)
  //     emailButton.click()
  //     expect(props.addGlobalMessage.mock.calls.length).toBe(1)
  // })

  // it('should handle email if events exist and are chosen and button is clicked', () => {
  //     const newProps = {...props, meetup: {...props.meetup, events: {...mocks.chosenEvent}}}
  //     const wrapper = shallow(<UnderlyingMeetup {...newProps}/>)
  //     const emailButton = wrapper.find('ProgressIcon')
  //     expect(emailButton).toHaveLength(1)
  //     emailButton.click()
  //     expect(props.sendMeetupEmails.mock.calls.length).toBe(1)
  // })

  it("should show edit meetup button and open form modal when clicked", () => {
    const wrapper = shallow(<UnderlyingMeetup {...props} />);
    expect(wrapper.state("editMeetupForm")).toBe(false);
    expect(wrapper.find('[aria-label="meetup-form"]')).toHaveLength(0);
    const editButton = wrapper.find('[aria-label="edit"]');
    expect(editButton).toHaveLength(1);
    editButton.simulate("click");
    expect(wrapper.state("editMeetupForm")).toBe(true);
    expect(wrapper.find('[aria-label="meetup-form"]')).toHaveLength(1);
  });
});

describe("Meetup tests where user is member", () => {
  const setup = () => {
    const newProps = { ...props, isUserMember: true };
    const wrapper = shallow(<UnderlyingMeetup {...newProps} />);
    return wrapper;
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should hide chat button on initial render", () => {
    const wrapper = setup();
    const button = wrapper.find('[aria-label="chat"]');
    expect(button).toHaveLength(0);
  });

  it("should show friend refresh button and send action", () => {
    const wrapper = setup();
    expect(props.getFriends.mock.calls.length).toBe(1);
    const button = wrapper.find('[aria-label="refresh-friends"]');
    expect(button).toHaveLength(1);
    button.simulate("click");
    expect(props.getFriends.mock.calls.length).toBe(2);
  });

  it("should hide chat if togglechat is called", () => {
    const wrapper = setup();
    expect(wrapper.find('[aria-label="meetup-chat"]')).toHaveLength(1);
    wrapper.setState({ showChat: false });
    expect(wrapper.find('[aria-label="meetup-chat"]')).toHaveLength(0);
  });

  it("should show new event button and open form modal when clicked", () => {
    const wrapper = setup();
    expect(wrapper.state("newMeetupEventForm")).toBe(false);
    expect(wrapper.find('[aria-label="add-event-form"]')).toHaveLength(0);
    const addButton = wrapper.find('[aria-label="add-event"]');
    expect(addButton).toHaveLength(1);
    addButton.simulate("click");
    expect(wrapper.state("newMeetupEventForm")).toBe(true);
    expect(wrapper.find('[aria-label="add-event-form"]')).toHaveLength(1);
  });

  it("should handle leave meetup", () => {
    const newProps = { ...props, isUserMember: true, user: mocks.user2 };
    const wrapper = shallow(<UnderlyingMeetup {...newProps} />);
    const leaveMeetupButton = wrapper.find('[aria-label="leave-meetup"]');
    expect(leaveMeetupButton).toHaveLength(1);
    leaveMeetupButton.simulate("click", { stopPropagation: () => undefined });
    expect(props.handleLeaveMeetup.mock.calls.length).toBe(1);
  });
});

describe("Meetup tests where user is not member", () => {
  const setup = () => {
    const newProps = { ...props, isUserMember: false };
    const wrapper = shallow(<UnderlyingMeetup {...newProps} />);
    return wrapper;
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should show join meetup button and send action on click", () => {
    const wrapper = setup();
    const button = wrapper.find('[aria-label="join-meetup"]');
    expect(button).toHaveLength(1);
    button.simulate("click");
    expect(props.handlePublicMeetupJoin.mock.calls.length).toBe(1);
  });

  it("should not show meetup chat", () => {
    const wrapper = setup();
    const chat = wrapper.find('[aria-label="meetup-chat"]');
    expect(chat).toHaveLength(0);
  });

  it("should not show member only buttons", () => {
    const wrapper = setup();
    const openChatButton = wrapper.find('[aria-label="chat"]');
    expect(openChatButton).toHaveLength(0);
    const leaveMeetupButton = wrapper.find('[aria-label="leave-meetup"]');
    expect(leaveMeetupButton).toHaveLength(0);
    const addEventButton = wrapper.find('[aria-label="leave-meetup"]');
    expect(addEventButton).toHaveLength(0);
  });
});

describe("Meetup tests where meetup is public", () => {
  const setup = () => {
    const wrapper = shallow(<UnderlyingMeetup {...props} />);
    return wrapper;
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should show PublicIcon button", () => {
    const wrapper = setup();
    const status = wrapper.find('[aria-label="meetup-type"]');
    expect(status).toHaveLength(1);
    expect(status.text()).toContain("Public");
  });
});

describe("Meetup tests where meetup is private", () => {
  const setup = () => {
    const newProps = { ...props, meetup: mocks.privateMeetup.uri2 };
    const wrapper = shallow(<UnderlyingMeetup {...newProps} />);
    return wrapper;
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should show PrivateIcon button", () => {
    const wrapper = setup();
    const status = wrapper.find('[aria-label="meetup-type"]');
    expect(status).toHaveLength(1);
    expect(status.text()).toContain("Private");
  });
});
