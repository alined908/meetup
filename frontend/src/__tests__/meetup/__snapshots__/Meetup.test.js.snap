// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meetup tests general renders without crashing given props 1`] = `
<div
  className="meetupWrapper"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <meta
      charSet="utf-8"
    />
    <title>
      Meetup
    </title>
  </HelmetWrapper>
  <MeetupTree
    initialized={false}
    meetup={
      Object {
        "categories": Array [
          Object {
            "api_label": "desserts",
            "id": 1,
            "image": null,
            "label": "Dessert",
          },
        ],
        "creator": Object {
          "avatar": null,
          "email": "daniel@gmail.com",
          "first_name": "Daniel",
          "id": 1,
          "last_name": "Lee",
        },
        "date": "2020-05-08",
        "events": Object {},
        "id": 1,
        "isMeetupEventsInitialized": true,
        "latitude": 34.228754,
        "location": "location",
        "longitude": -118.2351192,
        "members": Object {
          "1": Object {
            "admin": true,
            "ban": false,
            "id": 1,
            "user": Object {
              "avatar": null,
              "email": "daniel@gmail.com",
              "first_name": "Daniel",
              "id": 1,
              "last_name": "Lee",
            },
          },
          "2": Object {
            "admin": false,
            "ban": false,
            "id": 2,
            "user": Object {
              "avatar": null,
              "email": "example2@gmail.com",
              "first_name": "Bob",
              "id": 2,
              "last_name": "Jim",
            },
          },
        },
        "name": "Meetup",
        "notifs": 0,
        "public": true,
        "uri": "uri",
      }
    }
    sortEvents={[Function]}
  />
  <div
    className="meetupWrapperInner"
    id="head"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={3}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <div
          className="header elevate"
        >
          <WithStyles(ForwardRef(Typography))
            variant="h4"
          >
            Meetup
          </WithStyles(ForwardRef(Typography))>
          <div
            className="headerIcons"
          >
            <TodayIcon />
             
            Friday, May 8
          </div>
          <div
            aria-label="location"
            className="headerIcons"
          >
            <RoomIcon />
             
            location
          </div>
          <div
            aria-label="meetup-type"
            className="headerIcons"
          >
            <PublicIcon />
             Public
          </div>
          <div
            className="actions"
          >
            <WithStyles(ForwardRef(IconButton))
              edge="end"
              onClick={[Function]}
              style={
                Object {
                  "color": "rgba(10,10,10, .95)",
                }
              }
            >
              <MoreVertIcon />
            </WithStyles(ForwardRef(IconButton))>
            <WithStyles(ForwardRef(Menu))
              anchorEl={null}
              onClose={[Function]}
              open={null}
            >
              <WithStyles(ForwardRef(MenuItem))
                disabled={true}
                onClick={[Function]}
              >
                <WithStyles(ForwardRef(ListItemIcon))>
                  <EmailIcon
                    aria-label="email"
                    fontSize="small"
                    style={
                      Object {
                        "color": "black",
                      }
                    }
                  />
                </WithStyles(ForwardRef(ListItemIcon))>
                <WithStyles(ForwardRef(Typography))
                  noWrap={true}
                  variant="body2"
                >
                  Send Email
                </WithStyles(ForwardRef(Typography))>
              </WithStyles(ForwardRef(MenuItem))>
              <WithStyles(ForwardRef(MenuItem))
                onClick={[Function]}
              >
                <WithStyles(ForwardRef(ListItemIcon))>
                  <EditIcon
                    aria-label="edit"
                    fontSize="small"
                    style={
                      Object {
                        "color": "black",
                      }
                    }
                  />
                </WithStyles(ForwardRef(ListItemIcon))>
                <WithStyles(ForwardRef(Typography))
                  noWrap={true}
                  variant="body2"
                >
                  Edit Meetup
                </WithStyles(ForwardRef(Typography))>
              </WithStyles(ForwardRef(MenuItem))>
              <WithStyles(ForwardRef(MenuItem))
                onClick={[Function]}
              >
                <WithStyles(ForwardRef(ListItemIcon))>
                  <DeleteIcon
                    aria-label="delete"
                    color="secondary"
                    fontSize="small"
                  />
                </WithStyles(ForwardRef(ListItemIcon))>
                <WithStyles(ForwardRef(Typography))
                  noWrap={true}
                  variant="body2"
                >
                  Delete Meetup
                </WithStyles(ForwardRef(Typography))>
              </WithStyles(ForwardRef(MenuItem))>
            </WithStyles(ForwardRef(Menu))>
          </div>
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        id="Members"
        item={true}
        sm={6}
        xs={12}
      >
        <div
          className="headerIndented"
        >
          <div
            className="header elevate"
          >
            <WithStyles(ForwardRef(Typography))
              variant="h5"
            >
              Members
            </WithStyles(ForwardRef(Typography))>
            <WithStyles(ForwardRef(Button))
              aria-label="join-meetup"
              onClick={[Function]}
              style={
                Object {
                  "background": "#45B649",
                  "color": "white",
                }
              }
              variant="contained"
            >
              Join Meetup
            </WithStyles(ForwardRef(Button))>
          </div>
        </div>
        <div
          className="shell elevate"
        >
          <WithStyles(ForwardRef(List))
            className="shellList"
          >
            <Link
              key="1"
              to="/profile/1"
            >
              <WithStyles(ForwardRef(ListItem))>
                <WithStyles(ForwardRef(ListItemAvatar))>
                  <WithStyles(ForwardRef(Avatar))
                    src={null}
                  >
                    D
                    L
                  </WithStyles(ForwardRef(Avatar))>
                </WithStyles(ForwardRef(ListItemAvatar))>
                <WithStyles(ForwardRef(ListItemText))
                  primary="Daniel"
                  secondary={
                    <React.Fragment>
                      <ForwardRef(WithStyles)
                        color="inherit"
                        component="span"
                        variant="body2"
                      >
                        daniel@gmail.com 
                      </ForwardRef(WithStyles)>
                    </React.Fragment>
                  }
                />
                <WithStyles(ForwardRef(Tooltip))
                  title="Meetup Creator"
                >
                  <Component
                    height={24}
                    width={24}
                  />
                </WithStyles(ForwardRef(Tooltip))>
              </WithStyles(ForwardRef(ListItem))>
            </Link>
            <Link
              key="2"
              to="/profile/2"
            >
              <WithStyles(ForwardRef(ListItem))>
                <WithStyles(ForwardRef(ListItemAvatar))>
                  <WithStyles(ForwardRef(Avatar))
                    src={null}
                  >
                    B
                    J
                  </WithStyles(ForwardRef(Avatar))>
                </WithStyles(ForwardRef(ListItemAvatar))>
                <WithStyles(ForwardRef(ListItemText))
                  primary="Bob"
                  secondary={
                    <React.Fragment>
                      <ForwardRef(WithStyles)
                        color="inherit"
                        component="span"
                        variant="body2"
                      >
                        example2@gmail.com 
                      </ForwardRef(WithStyles)>
                    </React.Fragment>
                  }
                />
              </WithStyles(ForwardRef(ListItem))>
            </Link>
          </WithStyles(ForwardRef(List))>
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        sm={6}
        xs={12}
      >
        <div
          className="headerIndented"
        >
          <div
            className="header elevate"
          >
            <WithStyles(ForwardRef(Typography))
              variant="h5"
            >
              Friends
            </WithStyles(ForwardRef(Typography))>
            <WithStyles(ForwardRef(Tooltip))
              title="Refresh"
            >
              <WithStyles(ForwardRef(IconButton))
                aria-label="refresh-friends"
                onClick={[Function]}
                style={
                  Object {
                    "color": "black",
                  }
                }
              >
                <RefreshIcon />
              </WithStyles(ForwardRef(IconButton))>
            </WithStyles(ForwardRef(Tooltip))>
          </div>
        </div>
        <div
          className="shell elevate"
        >
          <WithStyles(ForwardRef(List))
            className="shellList"
          >
            <Connect(MeetupFriend)
              friend={
                Object {
                  "avatar": null,
                  "email": "example2@gmail.com",
                  "first_name": "Bob",
                  "id": 2,
                  "last_name": "Jim",
                }
              }
              isMember={true}
              isPast={false}
              key="1"
              uri="uri"
            />
          </WithStyles(ForwardRef(List))>
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <div
          className="hr"
          id="Events"
          style={
            Object {
              "fontSize": "1rem",
            }
          }
        >
          Events
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <div
          className="loading"
        >
          <WithStyles(ForwardRef(CircularProgress))
            size={30}
          />
        </div>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </div>
  <div
    className="addEvent"
  />
</div>
`;
